### PR TITLE
[FLINK-14066][python] Support to install pyflink in windows

### PR DESCRIPTION
**Pull request by dianfu**
## What is the purpose of the change

*This pull request adds support to install pyflink in windows.*

## Brief change log

  - *Fixes setup.py to allow pyflink could be installed in windows*

## Verifying this change

Test manually that pyflink could be installed in windows with both of the following commands:
- "python setup.py install"
- "python setup.py sdist; pip install .\dist\apache-flink-1.10.dev0.tar.gz"

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-python/setup.py [hunk 1]
```diff
@@ -108,8 +119,13 @@
             print(incorrect_invocation_message, file=sys.stderr)
             sys.exit(-1)
 
-        if getattr(os, "symlink", None) is not None:
+        try:
```
Comment (HuangXingBo) [inline flink-python/setup.py]: What's the reason of this change? I believe there must be some reason that I don't know.

Comment (dianfu) [reply flink-python/setup.py]: There are cases the os support symlink however the disk doesn't support symlink. This is found when we test this PR on a windows virtual machine in which the disk is a mounted virtual disk.

File: flink-python/setup.py [hunk 2]
```diff
@@ -75,8 +85,9 @@
             print("Temp path for symlink to parent already exists {0}".format(TEMP_PATH),
                   file=sys.stderr)
             sys.exit(-1)
-
-        FLINK_HOME = os.path.abspath("../build-target")
+        java_version = VERSION.replace(".dev0", "-SNAPSHOT")
```
Comment (hequn8128) [inline flink-python/setup.py]: How about renaming it to flink_version? I was first confused that it is the version of jvm.
